deepchem/models/torch_models/infograph.py

```diff
@@ -6,6 +6,269 @@
+                torch.nn.init.xavier_uniform_(m.weight.data)
+                if m.bias is not None:
+                    m.bias.data.fill_(0.0)
+
+    def forward(self, data):
+        y, M = self.encoder(data)
+        g_enc = self.global_d(y)
+        l_enc = self.local_d(M)
+        return g_enc, l_enc
+
+
+class InfoGraphModel(ModularTorchModel):
+    """
+    InfoGraph is a graph convolutional model for unsupervised graph-level representation learning. The model aims to maximize the mutual information between the representations of entire graphs and the representations of substructures of different granularity.
+
+    The unsupervised training of InfoGraph involves two encoders: one that encodes the entire graph and another that encodes substructures of different sizes. The mutual information between the two encoder outputs is maximized using a contrastive loss function.
+    The model randomly samples pairs of graphs and substructures, and then maximizes their mutual information by minimizing their distance in a learned embedding space.
+
+    This can be used for downstream tasks such as graph classification and molecular property prediction.It is implemented as a ModularTorchModel in order to facilitate transfer learning.
+
+    References
+    ----------
+    .. [1] F.-Y. Sun, J. Hoffmann, V. Verma, and J. Tang, “InfoGraph: Unsupervised and Semi-supervised Graph-Level Representation Learning via Mutual Maximization.” arXiv, Jan. 17, 2020. http://arxiv.org/abs/1908.01000
+
+    Parameters
+    ----------
+    num_features: int
+        Number of node features for each input
+    edge_features: int
+        Number of edge features for each input
+    embedding_dim: int
+        Dimension of the embedding
+    num_gc_layers: int
+        Number of graph convolutional layers
+    prior: bool
+        Whether to use a prior expectation in the loss term
+    gamma: float
+        Weight of the prior expectation in the loss term
+    measure: str
+        The divergence measure to use for the unsupervised loss. Options are 'GAN', 'JSD',
+        'KL', 'RKL', 'X2', 'DV', 'H2', or 'W1'.
+    average_loss: bool
+        Whether to average the loss over the batch
+    """
+
+    def __init__(self,
+                 num_features,
+                 embedding_dim,
+                 num_gc_layers=5,
+                 prior=True,
+                 gamma=.1,
+                 measure='JSD',
+                 average_loss=True,
+                 **kwargs):
+        self.num_features = num_features
+        self.embedding_dim = embedding_dim * num_gc_layers
+        self.num_gc_layers = num_gc_layers
+        self.gamma = gamma
+        self.prior = prior
+        self.measure = measure
+        self.average_loss = average_loss
+        self.components = self.build_components()
+        self.model = self.build_model()
+        super().__init__(self.model, self.components, **kwargs)
+
+    def build_components(self) -> dict:
```

Comment:
Can you add a docstring explaining the components of the model? Model components should be documented for modular torch models in general